soda-sqlserver/src/soda_sqlserver/common/data_sources/sqlserver_data_source.py

```diff
@@ -304,3 +313,19 @@ def map_test_sql_data_type_to_data_source(self, source_data_type: SqlDataType) -
         if sql_data_type.name == "varchar":
             sql_data_type.character_maximum_length = self.default_varchar_length()
         return sql_data_type
+
+    def is_same_soda_data_type(self, expected: SodaDataTypeName, actual: SodaDataTypeName) -> bool:
+        found_synonym = False
+        synonym_correct = False
+        if expected == SodaDataTypeName.TEXT or actual == SodaDataTypeName.VARCHAR:
```

Comment:
Think there is an error here?


```suggestion
        if expected == SodaDataTypeName.TEXT or expected == SodaDataTypeName.VARCHAR:
```

It's similar pattern as above, so might be replaced with utility method?